libcloud/compute/drivers/oneandone.py

```diff
@@ -0,0 +1,2263 @@
+
+        return self._to_node(response.object)
+
+    def ex_remove_server_ip(self, server_id, ip_id, keep_ip=None):
+        """
+        Removes an IP address from the server
+
+        :param server_id: Id of the server
+        :rtype: ``str``
+
+        :param ip_id: ID of the IP address
+        :rtype: ``str``
+
+        :param keep_ip: Indicates whether IP address will be removed from
+        the Cloud Panel
+        :rtype: ``boolean``
+
+        :return: ``Node`` instance
+        :rtype: ``Node``
+        """
+
+        body = {}
+        if keep_ip is not None:
+            body['keep_ip'] = keep_ip
+
+        response = self.connection.request(
+            action='servers/%s/ips/%s' % (server_id, ip_id),
+            data=body,
+            method='DELETE'
+        )
+
+        return self._to_node(response.object)
+
+    def ex_get_server_firewall_policies(self, server_id, ip_id):
+        """
+        Gets a firewall policy of attached to the server's IP
+        :param server_id: Id of the server
+        :rtype: ``str``
+
+        :param ip_id: ID of the IP address
+        :rtype: ``str``
+
+        :return: IP address object
+        :rtype: ``dict``
+        """
+
+        response = self.connection.request(
+            action='/servers/%s/ips/%s/firewall_policy' % (server_id, ip_id),
+            method='GET'
+        )
+
+        return response.object
+
+    def ex_remove_server_firewall_policy(self, server_id, ip_id):
+        """
+        Removes a firewall policy from server's IP
+
+        :param server_id: Id of the server
+        :rtype: ``str``
+
+        :param ip_id: ID of the IP address
+        :rtype: ``str``
+
+        :return: ``Node`` instance
+        :rtype: ``Node``
+        """
+        response = self.connection.request(
+            action='/servers/%s/ips/%s/firewall_policy' % (server_id, ip_id),
+            method='DELETE'
+        )
+
+        return self._to_node(response.object)
+
+    def ex_add_server_firewall_policy(self, server_id, ip_id, firewall_id):
+        """
+        Adds a firewall policy to the server's IP address
+
+        :param server_id: Id of the server
+        :rtype: ``str``
+
+        :param ip_id: ID of the IP address
+        :rtype: ``str``
+
+        :param firewall_id: ID of the firewall policy
+        :rtype: ``str``
+
+        :return: ``Node`` instance
+        :rtype: ``Node``
+        """
+        body = {
+            "id": firewall_id
+        }
+        response = self.connection.request(
+            action='/servers/%s/ips/%s/firewall_policy' % (server_id, ip_id),
+            data=body,
+            method='POST'
+        )
+
+        return self._to_node(response.object)
+
+    """
+    Firewall Policy operations
+    """
+
+    def ex_create_firewall_policy(self, name, rules, description=None):
+        """
+        Creates a firewall Policy
+        :param name:
+        :param description:
+        :param rules:
+        :rtype 'dict':  [
```

Comment:
I'm not sure what do you mean by this. I wanted to give an example of what dict needed to be passed to the function. I can easily remove it.